Modules/signalmodule.c

```diff
@@ -1760,11 +1760,22 @@ _PySignal_Fini(void)
 }
 
 
-/* Declared in pyerrors.h */
-int
-PyErr_CheckSignals(void)
+/* Subroutine of the PyErr_CheckSignals family.  Does all the work
+   that needs an attached thread state.  When called, 'tstate' must
+   be the thread state for the current thread, and it must be attached.  */
+static int
+check_signals_attached(PyThreadState *tstate, bool cycle_collect)
 {
-    PyThreadState *tstate = _PyThreadState_GET();
+#if defined(Py_REMOTE_DEBUG) && defined(Py_SUPPORTS_REMOTE_DEBUG)
+    _PyRunRemoteDebugger(tstate);
+#endif
+
+    /* It is necessary to repeat all of the checks of global flags
+       that were done in _PyErr_CheckSignalsNoGIL.  At the time of
+       those checks, we might not have held the GIL; in between those
+       checks and when we acquired the GIL, some other thread may have
+       processed the events that were flagged.  Since we now hold the
```

Comment:
No other *thread* could have processed the events (only the main thread of the main interpreter can) but new events might have been added.

```suggestion
       checks and when we acquired the GIL, events may have been
       flagged or unflagged.  Since we now hold the
```

Reply:
Will correct.